torch/csrc/cuda/Module.cpp

```diff
@@ -767,6 +769,8 @@ PyObject* THCPModule_memorySnapshot(PyObject* _unused, PyObject* noargs) {
     throw std::runtime_error("unreachable");
   };
 
+  auto converter = c10::cuda::CUDACachingAllocator::getApproxClockConverter();
```

Comment:
Can we move this conversion into the `::snapshot()` function? Doing it this way means that we do not have expose `getApproxClockConverter`, which is complicated since the other allocators do not support it.

The output of snapshot already copies `TraceEntry` outside of the buffer before it returns, so it can then mutate the te.time_ entry in place to turn it into unix time. (If approx_time_t isn't just uint64_t we can add a union to TraceEntry)

Reply:
Do you know how long it takes for the start of `:snapshot` to get to this point? The `ApproximateClockToUnixTimeConverter`'s init needs to be a good time difference before calling `makeConverter()`, otherwise the scaling factor when converting from tsc to ns is a bit off. I will give it a try from the `::snapshot()` function.

Sorry didn't understand the second part, do you mean that I should convert to unix time sooner in the Caching Allocator?

Reply:
@zdevito I was able to resolve the comment, and moved the converter to the snapshot function, removed need to support in other allocators, and used a union for allow storing in place from approx_time_t to time_t.